Sources/MapboxCoreNavigation/PassiveLocationManager.swift

```diff
@@ -217,18 +218,25 @@ open class PassiveLocationManager: NSObject {
         case .some(_):
             break
         }
+        
+        let actualSpeed: Measurement<UnitSpeed>? = (lastLocation.speed >= 0) ? Measurement(value: lastLocation.speed, unit: .metersPerSecond) : nil
```

Comment:
The test app is also using the `CLLocation.speed` to get the current speed, and rounded it, formatted it to the desired  unit to display. Because I'm not sure how the design should look like. Here I only converted it to the current speed unit and checked whether it's valid or not.

Reply:
Consider keeping the current speed as a `CLLocationSpeed` (in meter per second) and converting to the local speed unit at the UI level rather than in the location manager. A speed limit is inherently associated with a particular unit, whereas the current speed is just a measurement, which is valid regardless of the unit it’s expressed in. `SpeedLimitView` would convert the speed to the same unit as the speed limit for consistency, but perhaps there’s some other use case for the current speed that doesn’t require the same units.